simulator-starter/src/test/java/com/consol/citrus/simulator/service/ScenarioExecutionRepositoryTest.java

```diff
@@ -0,0 +1,250 @@
+ * limitations under the License.
+ */
+package com.consol.citrus.simulator.service;
+
+import com.consol.citrus.DefaultTestCase;
+import com.consol.citrus.simulator.SimulatorAutoConfiguration;
+import com.consol.citrus.simulator.config.SimulatorConfigurationProperties;
+import com.consol.citrus.simulator.model.Message.Direction;
+import com.consol.citrus.simulator.model.ScenarioExecution;
+import com.consol.citrus.simulator.model.ScenarioExecution.Status;
+import com.consol.citrus.simulator.model.ScenarioExecutionFilter;
+import com.consol.citrus.simulator.repository.ScenarioExecutionRepository;
+import org.junit.runner.RunWith;
+import org.testng.Assert;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.testng.annotations.Test;
+import java.util.*;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@ContextConfiguration(classes= {SimulatorAutoConfiguration.class})
+public class ScenarioExecutionRepositoryTest {
+    
+    private static final String TEST_SCENARIO= "test-scenario";
+
+    private static final SimulatorConfigurationProperties PROPERTIES = new SimulatorConfigurationProperties();
+    
+    private static final QueryFilterAdapterFactory queryFilterAdapterFactory = new QueryFilterAdapterFactory(PROPERTIES);
+    
+    private static final String PAYLOAD = "test-payload";
+    
+    private static final String IN_HEADER_NAME1 = "IH1";
+    private static final String IN_HEADER_NAME2 = "IH2";
+    
+    private static final String OUT_HEADER_NAME1 = "OH1";
+    private static final String OUT_HEADER_NAME2 = "OH2";
+        
+    @Autowired
+    private ActivityService activityService;
+    
+    @Autowired
+    private ScenarioExecutionRepository scenarioExecutionRepository;
+
+    @Test
+    void testFindByScenarioName() {
+        String inUid = UUID.randomUUID().toString();
+        String outUid = UUID.randomUUID().toString();
+        
+        String uniqueScenarioName = TEST_SCENARIO + UUID.randomUUID().toString();
+        createTestScenarioExecution(uniqueScenarioName, inUid, PAYLOAD, outUid, PAYLOAD, Status.SUCCESS);
+        
+        ScenarioExecutionFilter scenarioExecutionFilter = new ScenarioExecutionFilter();
+        scenarioExecutionFilter.setScenarioName(uniqueScenarioName);
+        
+        ScenarioExecutionFilter queryFilter = queryFilterAdapterFactory.getQueryAdapter(scenarioExecutionFilter);
+        
+        Assert.assertEquals(1, scenarioExecutionRepository.find(queryFilter).size());
+    }
+    
+    @Test
+    void testFindByScenarioStatus() {
+        String inUid = UUID.randomUUID().toString();
+        String outUid = UUID.randomUUID().toString();
+        
+        String uniqueScenarioName = TEST_SCENARIO + UUID.randomUUID().toString();
```

Comment:
I would prefer static uuids in tests. it's more reproducible: https://duckduckgo.com/?q=random+uuid&ia=answer. up to you :)

Reply:
I don't think that this would make the code more readable. Also. Putting static UUID holds the risk of copy/pasting the same UUID into other methods making them not unique any more. You end up looking into issue resulting from non unique ids and these  could be quit cumbersome to solve.

Reply:
as said, ok for me.